bin/varnishd/cache/cache_backend.c

```diff
@@ -271,6 +271,9 @@ vbe_dir_getfd(VRT_CTX, struct worker *wrk, VCL_BACKEND dir, struct backend *bp,
 		return (NULL);
 	}
 
+	vbe_connwait_fini(cw);
```

Comment:
I am pretty sure that I stared at this for long in the course of #4030 and there was a good reason why we can not move the fini. If you insist, I need to stare again

Reply:
That was true for #4030 , but these commits (from this PR) made it possible to simplify things further:
- https://github.com/varnishcache/varnish-cache/pull/4154/commits/687011799fa72b30cb2d08c25a77eff13cb6538d
- https://github.com/varnishcache/varnish-cache/pull/4154/commits/fe747ac16ef15ef228a5f245fab5bd6372c01f30

The `.max_connections` bug was present before #4030 . Fixing this bug enable us to make the connection queue implementation more compact.

Reply:
This has been removed from this PR, I will push a separate PR for the other commits.